account-model/src/account/mod.rs

```diff
@@ -0,0 +1,56 @@
+use std::convert::TryFrom;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum Classification {
+    PERSONAL,
+    BUSINESS,
+}
+
+impl Classification {
+    pub fn value(&self) -> &'static str {
+        match *self {
+            Classification::PERSONAL => "Personal",
+            Classification::BUSINESS => "Business",
+        }
+    }
+}
+
+impl TryFrom<&str> for Classification {
+    type Error = String;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        match input {
+            "Personal" => Ok(Classification::PERSONAL),
+            "Business" => Ok(Classification::BUSINESS),
+            _ => Err(format!("{} is not classification variant", input))
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::Classification;
+
+    macro_rules! parametrised {
+        ($($name:ident: $value:expr,)*) => {
+        $(
+            #[test]
+            fn $name() {
+                let (input, expected, runnable) = $value;
+                assert_eq!(expected, runnable(input));
+            }
+        )*
+        }
+    }
+
+    parametrised! {
+        personal_to_value: (Classification::PERSONAL, "Personal", |c: Classification| c.value()),
+        business_to_value: (Classification::BUSINESS, "Business", |c: Classification| c.value()),
+    }
+
+    // parametrised! {
```

Comment:
przerasta mnie to jak na dziś, dostaje info że 
```
error[E0277]: the trait bound `std::result::Result<account::Classification, _>: std::convert::From<&str>` is not satisfied
  --> account-model/src/account/mod.rs:52:86
   |
52 |         personal_from_string: ("Personal", Ok(Classification::PERSONAL), |i: &str| i.into()),
   |                                                                                      ^^^^ the trait `std::convert::From<&str>` is not implemented for `std::result::Result<account::Classification, _>`
   |
   = note: required because of the requirements on the impl of `std::convert::Into<std::result::Result<account::Classification, _>>` for `&str`
```

widziałem u ciebie w testach `let message: Message = (id, request).into();` więc może tutaj to jest kwestia makro generującego testy ? nie wiem, poddaje się na dziś. a chciałem tylko zawołać `Classification.try_from("Personal")` ale widać to jest za proste.

gdybym zastąpił test
```
parametrised! {
        personal_from_string: ("Personal", Classification::PERSONAL, |i: &str| i.into().unwrap()),
```
dostaje
```
error[E0282]: type annotations needed
  --> account-model/src/account/mod.rs:52:82
   |
52 |         personal_from_string: ("Personal", Classification::PERSONAL, |i: &str| i.into().unwrap()),
   |                                                                                  ^^^^ cannot infer type for `T`
   |
   = note: type must be known at this point
```
więc na pewno to jest kwestia tej lambdy, ale nie wiem wtedy jak przetestować input == `unknown`

te traity chyba mnie nie interesują :/
https://doc.rust-lang.org/std/convert/trait.From.html
https://doc.rust-lang.org/std/convert/trait.FromStr.html